packages/aws-cdk/lib/api/aws-auth/awscli-compatible.ts

```diff
@@ -46,10 +45,9 @@ export class AwsCliCompatible {
     ];
 
     if (await fs.pathExists(credentialsFileName())) {
-      sources.push(() => new AWS.SharedIniFileCredentials({ profile, filename: credentialsFileName(), httpOptions, tokenCodeFn }));
-    }
-
-    if (await fs.pathExists(configFileName())) {
+      // Force reading the `config` file if it exists by setting the appropriate
+      // environment variable.
+      await forceSdkToReadConfigIfPresent();
       sources.push(() => new AWS.SharedIniFileCredentials({ profile, filename: credentialsFileName(), httpOptions, tokenCodeFn }));
```

Comment:
I remember we had the `if` branch here because the `SharedIniFileCredentials` would fail if the file did not actually exist... Has this been fixed?

Reply:
That is not this `if`, that's a different one (inside `forceSdkToReadConfigIfPresent`).